docs/sphinx/source/reference/sql_commands/DQL/WITH.diagram

```diff
@@ -1,29 +1,49 @@
 Diagram(
     Stack(
         Terminal('WITH'),
-            OneOrMore(
-                Sequence(
-                    NonTerminal('cteName'),
-                    Optional(
-                        Sequence(
-                            Terminal('('),
-                            OneOrMore(
-                                NonTerminal('columnAlias'),
-                                Terminal(','),
-                            ),
-                            Terminal(')'),
+        Optional(
+            Sequence(
+                Terminal('RECURSIVE'),
+                Optional(
+                    Sequence(
+                        Terminal('('),
+                        Terminal('TRAVERSAL'),
+                        Terminal('='),
+                        Choice(0,
+                            Terminal('ANY'),
+                            Terminal('PREORDER'),
```

Comment:
I think I would suggest a parenthesis-less syntax:
```
WITH RECURSIVE cte TRAVERSAL ORDER PRE-ORDER AS ...
```

Or 
```
WITH RECURSIVE cte AS (
SELECT ...
UNION ALL
SELECT ...
)
TRAVERSAL ORDER PRE-ORDER

SELECT * FROM cte
```

with a preference for the second proposal since it aligns with the SEARCH clause in terms of positioning.

Reply:
The railroad diagram looks like this now:

<img width="811" height="459" alt="image" src="https://github.com/user-attachments/assets/2b3f0be6-3995-40f4-9027-a84668fa10fb" />